controlled-vocabulary/src/main/java/uk/ac/ebi/uniprot/cv/keyword/impl/KeywordEntryImpl.java

```diff
@@ -0,0 +1,195 @@
+package uk.ac.ebi.uniprot.cv.keyword.impl;
+
+import uk.ac.ebi.uniprot.cv.keyword.GeneOntology;
+import uk.ac.ebi.uniprot.cv.keyword.Keyword;
+import uk.ac.ebi.uniprot.cv.keyword.KeywordEntry;
+import uk.ac.ebi.uniprot.cv.keyword.KeywordStatistics;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class KeywordEntryImpl implements KeywordEntry {
+
+    private Keyword keyword;
+    private String definition;
+    private List<String> synonyms;
+    private List<GeneOntology> geneOntologies;
+    private Set<KeywordEntry> parents;
+    private List<String> sites;
+    private Keyword category;
+    private List<KeywordEntry> children;
+    private KeywordStatistics statistics;
+
+
+    public KeywordEntryImpl() {
+        children = new ArrayList<>();
+        parents = new HashSet<>();
+    }
+
+    public KeywordEntryImpl(Keyword keyword, String definition, List<String> synonyms,
+                            List<GeneOntology> geneOntologies, Set<KeywordEntry> parents, List<String> sites,
+                            Keyword category, List<KeywordEntry> children, KeywordStatistics statistics) {
+        this.keyword = keyword;
+        this.definition = definition;
+        this.synonyms = synonyms;
+        this.geneOntologies = geneOntologies;
+        this.parents = parents;
+        this.sites = sites;
+        this.category = category;
+        this.children = children;
+        this.statistics = statistics;
+    }
+
+    public void setDefinition(String definition) {
+        this.definition = definition;
+    }
+
+
+    public void setSynonyms(List<String> synonyms) {
```

Comment:
Do you think we should stick to using Builders for setting values, e.g., using the `from` method like we did with the UniProt domain model?